packages/@lwc/synthetic-shadow/src/faux-shadow/slot.ts

```diff
@@ -83,9 +89,20 @@ function getFilteredSlotFlattenNodes(slot: HTMLElement): Node[] {
     );
 }
 
-export function assignedSlotGetterPatched(this: Element): HTMLSlotElement | null {
+export function assignedSlotGetterPatched(this: Element | Text): HTMLSlotElement | null {
     const parentNode = parentNodeGetter.call(this);
 
+    // use original assignedSlot if parent has a native shdow root
+    if (parentNode instanceof Element) {
+        const sr = parentNode.shadowRoot;
+        if (sr && isInstanceOfNativeShadowRoot(sr)) {
```

Comment:
`isInstanceOfNativeShadowRoot` can handle `null` so we don't need the extra check.
```suggestion
        if (isInstanceOfNativeShadowRoot(sr)) {
```